### PR TITLE
refactored tests using jest

### PR DESCRIPTION
**Pull request by fkropfhamer**
Refactored unittest using the test-framework jest for more legibility and better error handling.

### CODE REVIEW

File: 1-math/package.json [hunk 1]
```diff
@@ -2,9 +2,12 @@
...
   "scripts": {
-    "test": "node mat_test.js"
+    "test": "jest"
   },
   "repository": "mimuc/cg1-ss20",
   "author": "LMU Munich MIMUC CG1 Team",
-  "license": "GPL-3.0+"
+  "license": "GPL-3.0+",
+  "devDependencies": {
+    "jest": "^25.4.0"
```
Comment (changkun) [inline 1-math/package.json:11]: The goal of the original test file is to use pure JS and add test case more convenient without redundantly copy them and offering a way to students to test their implementation. Therefore all naive test cases are grouped, and the test is executed by looping all data.

Introducing dependencies are not intended initially while designing Assignment 1.
Although using a testing framework is good, but I treat this as a tradeoff, and it is clearly shown in the PR, see the lengthy `package-lock.json`.

It leads to a situation where the current error message is worse than what Jest is offered, but to some newcomers, the error message might (or might not) be more complicated to them.

The questions from my side are:

- In which case, the error message is not user friendly to you (and potentially to other students)?
- Is it possible for you to implement those Jest handy functions, without introducing any dependent libraries? e.g. implement `test()`, `expect()`, `toBeEqual()` and etc.
- Will you consider improve the error message directly by better error handling other than using a library?

File: 1-math/mat.js [hunk 2]
```diff
@@ -54,7 +54,7 @@ class Vector3 {
    * @param {Vector3} w is a 3 dimensional vector
    */
   angle(w) {
-    // TODO: Implement angule of this and w: ∠(this, w)
+    // TODO: Implement angle of this and w: ∠(this, w)
```
Comment (changkun) [inline 1-math/mat.js:57]: A typo fix can be sent via a different pull request :)

**Review by changkun — COMMENTED** [review]
Thank you for your contribution! I've reviewed your change list and please find the comments below :)

**Comment by fkropfhamer** [comment]
- My problem was in the matrix multiply function. In my incorrect implementation I switched the rows of the solution matrix. The Test showed me the solution was incorrect, but the Error message showed expected: ... xs: [Array] ... got: [1,2,3,5,6,7,8] so I could not see the mistake and had to console.log it myself. 
- If the goal is to use pure js is understand the problem, I was just confused. At least it was a good exercise.

**Comment by changkun** [comment]
> * My problem was in the matrix multiply function. In my incorrect implementation, I switched the rows of the solution matrix. The Test showed me the solution was incorrect, but the Error message showed expected: ... xs: [Array] ... got: [1,2,3,5,6,7,8] so I could not see the mistake and had to console.log it myself.

Very good experience report. I didn't know this before. I think this could be addressed by a better logger in printing values.

> * If the goal is to use pure js is understand the problem, I was just confused. At least it was good exercise.

Thanks and sorry for the frustrating error messaging. A further PR regarding error logging is welcome :)
